keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -91,32 +85,52 @@ def __init__(
     def get_random_transformation(
         self, image=None, label=None, bounding_box=None, **kwargs
     ):
-        crop_area_factor = self.crop_area_factor()
+        zoom_factor = self.zoom_factor()
         aspect_ratio = self.aspect_ratio_factor()
 
-        new_height = tf.clip_by_value(
-            tf.sqrt(crop_area_factor / aspect_ratio), 0.0, 1.0
-        )  # to avoid unwanted/unintuitive effects
-        new_width = tf.clip_by_value(tf.sqrt(crop_area_factor * aspect_ratio), 0.0, 1.0)
+        if image.shape.rank == 3:
```

Comment:
you can achieve this behavior using negative indices:

original_heights = image.shape[-2] I believe should work!  examples of this are available in RandomCrop/RandomHeight